ehitchhiking-ui/src/app/shared/components/rate-passengers-modal/rate-passengers-modal.component.ts

```diff
@@ -2,60 +2,96 @@ import {Component, Inject, OnInit} from '@angular/core';
+import {LoaderSize} from '@shared/enums/pre-loader-sizes';
+import {BlacklistedUser} from '@shared/components/rate-passengers-modal/user';
 
 @Component({
 	selector: 'rate-passengers-modal',
 	templateUrl: './rate-passengers-modal.component.html',
 	styleUrls: ['./rate-passengers-modal.component.sass'],
 })
-export class RatePassengersModalComponent implements OnInit{
-  users: User[] = [];
-  UserState = UserState;
-
-  constructor(public dialogRef: MatDialogRef<RatePassengersModalComponent>, @Inject(MAT_DIALOG_DATA) public data: any) {
-  }
-
-  ngOnInit(): void {
-    // here should be just GET 'info' about trip
-    if (this.data.dataKey === UserState.Driver){
-      this.users = [
-        {id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false},
-        {id: 1, name: 'Egor', rating: 0, isAddedToBlackList: false},
-        {id: 2, name: 'Semen', rating: 0, isAddedToBlackList: false},
-        {id: 3, name: 'Fedor', rating: 0, isAddedToBlackList: false},
-      ];
-    }
-    else{
-      this.users = [{id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false}];
+export class RatePassengersModalComponent implements OnInit {
+	users: RatedUser[] = [];
+	UserState = UserState;
+	idTripDriver = 12;
+	idTripPassenger = 4;
+	loading = true;
+	loaderSize: LoaderSize = LoaderSize.Large;
+	constructor(
+		public dialogRef: MatDialogRef<RatePassengersModalComponent>,
+		@Inject(MAT_DIALOG_DATA) public data: any,
+		private apiRatePassengersService: RatePassengersApiService
+	) {}
+
+	ngOnInit(): void {
+    switch(this.data.dataKey) {
+      case UserState.Passenger:
+        this.loadDriversList();
+        break;
+      case UserState.Driver:
+        this.loadPassengersList();
+        break;
+      default:
+        break;
     }
+	}
 
+	loadPassengersList():void {
+		this.loading = true;
+		this.apiRatePassengersService.getTripPassengers(this.idTripPassenger).then((data: RatedUser[]) => {
+			this.users = data;
+		}).finally(() => this.loading = false);
+	}
 
-  }
+	loadDriversList(): void {
+		this.loading = true;
+		this.apiRatePassengersService.getTripDriver(this.idTripDriver).then((data: RatedUser[]) => {
+			this.users = data;
+		}).finally(() => this.loading = false);
+	}
 
-  rateUser(clickObj: StarClickMeta): void {
-		const user = this.users.find((i: User) => i.id === clickObj.itemId);
-		if (!!user) {
-			user.rating = clickObj.rating;
-		}
+	rateUser(clickObj: StarClickMeta): void {
+	  this.users.forEach((i: RatedUser) => {
+	    i.id === clickObj.itemId ? (i.rate = clickObj.rating) : '';
+	    return i;
+	  });
 	}
 
 	exitTrip(): void {
-		this.dialogRef.close();
+		const users = [];
+		const blockedUsers= [];
+		this.users.forEach((user) => {
+			users.push({
+				idDriver: user.id,
+				rate: user.rate,
+			});
+			blockedUsers.push({
+				idPass: user.id,
+				rate: user.rate,
+			});
+		});
+		if (this.data.dataKey === UserState.Passenger) {
+			this.apiRatePassengersService.addRateDriver(users);
```

Comment:
pls show pre-loader during loading

You should hide loading when BOTH request will be finished. You should close modal when BOTH request will be finished successfully

Reply:
done